quickwit/quickwit-indexing/src/split_store/local_split_store.rs

```diff
@@ -134,12 +134,19 @@ impl SplitFolder {
     }
 }
 
-fn split_id_from_split_folder(dir_path: &Path) -> Option<&str> {
-    if !dir_path.is_dir() {
-        return None;
+fn split_id_from_split_folder(dir_path: &Path) -> io::Result<Option<&str>> {
+    let metadata = dir_path.metadata()?;
+
+    if !metadata.is_dir() {
```

Comment:
Now, we're going to throw an error about the file name when the issue is that we're dealing with a file. I'd deal with the path not found or not a dir outside of this function:
```rust
for dir_entry in ... {
  let dir_path: PathBuf = dir_entry.path();
  let metadata = tokio::fs::metadata(&dir_path).await?;
  
  if metadata.is_file() {
    warn!("Unexpected file found in split cache directory: `{}`.", dir_path.display());
    continue;
  }
  let split_id = split_id_from_split_folder(&dir_path).ok_or_else(|| ...)?;
}

Reply:
Sounds good to me, have moved it out